src/app/components/approval-requests/approval-requests.component.ts

```diff
@@ -0,0 +1,66 @@
+import { Component, OnInit, NgZone } from '@angular/core';
+import { Router } from '@angular/router';
+
+import { ChromeMessageType } from 'src/app/core/enums/chrome-message-type';
+import { ToastrService } from 'ngx-toastr';
+import { VaultService } from 'src/app/core/services/vault/vault.service';
+
+@Component({
+  selector: 'app-approval-requests',
+  templateUrl: './approval-requests.component.html',
+  styleUrls: ['./approval-requests.component.scss']
+})
+export class ApprovalRequestsComponent implements OnInit {
+  public type: string;
+  public from: string;
+  public requestsCount: number;
+
+  constructor(
+    private router: Router,
+    private toastr: ToastrService,
+    private vaultService: VaultService,
+    private zone: NgZone) { }
+
+  ngOnInit() {
+    this.getApprovalRequestsCount();
+    this.getApprovalRequest();
+  }
+
+  sendRequestResponse(approved: boolean) {
+    if (approved) {
+      this.vaultService.addWhitelistedDomain(this.type, this.from);
+      this.toastr.success('Request approved!', null, {timeOut: 1000});
+    } else {
+      this.toastr.info('Request cancelled!', null, {timeOut: 1000});
+    }
+
+    chrome.runtime.sendMessage({type: ChromeMessageType.SendApprovalRequestResponse, success: approved}, (response) => {
+      this.zone.run(() => {
+        this.requestsCount = response.approvalRequestsCount;
+        this.getApprovalRequest();
+      });
+    });
+  }
+
+  private getApprovalRequest() {
+    chrome.runtime.sendMessage({type: ChromeMessageType.GetApprovalRequest}, (response) => {
+      this.zone.run(() => {
+        if (response.success) {
+          const request = response.approvalRequest;
+          this.type = request.type;
+          this.from = request.from;
+        } else {
+          this.router.navigate(['home']);
+        }
+      });
+    });
+  }
+
+  private getApprovalRequestsCount() {
+    chrome.runtime.sendMessage({type: ChromeMessageType.ApprovalRequestsCount}, (response) => {
+      this.zone.run(() => {
+        this.requestsCount = response.approvalRequestsCount;
+      });
+    });
+  }
```

Comment:
We need the count because you can have more than one approval requests at a time and we are displaying it to the user in the UI, the same way we are displaying the count for the pending signing requests. When you approve or deny request, the count is updated and a notification message is shown.